portindex2json/portindex2json.tcl

```diff
@@ -64,7 +64,22 @@ proc remove_extra_braces {text} {
     return [regsub -all {[\{\}]} $text ""]
 }
 
-set fd [open [lindex $argv 0] r]
+for {set i 0} {$i < $argc} {incr i} {
+    set arg [lindex $argv $i]
+    switch -regex -- $arg {
+        {^-.+} {
+            if {$arg eq "-info"} {
+                incr i
+                set argument [split [lindex $argv $i] =]
+                set json_info([lindex $argument 0]) [::json::write string [lindex $argument 1]]
+            }
+        }
+        {$.PortIndex|i} {
+            set fd [open [lindex $argv $i] r]
+        }
```

Comment:
What happens now if:
* you call `portindex2json.tcl invalidparameter`
* you provide a filename that doesn't exist
?

I would assume that the command should probably gracefully fail with a nice/helpful error message. That's not super important since that was nearly the case before, just potentially nice to think about.

Reply:
> What happens now if:
> 
> * you call `portindex2json.tcl invalidparameter`

It won't read the invalid parameters. It will read only `-info`

> * you provide a filename that doesn't exist
>   ?

It throws an error:
```
couldn't open "macports-ports/portindex": no such file or directory
```

Reply:
I'm getting lots of other errors:
```
can't read "fd": no such variable
    while executing
"chan configure $fd -encoding utf-8"
    (file "portindex2json.tcl" line 84)
```
You should exit early (print invocation and optionally a message saying that filename is missing) rather than trying to proceed even when the filename wasn't set.